fundingmanager.go

```diff
@@ -2216,6 +2422,86 @@ func copyPubKey(pub *btcec.PublicKey) *btcec.PublicKey {
 	}
 }
 
+// saveOpenChanAnnPref saves an open channel's announcement preference.
+func (f *fundingManager) saveOpenChanAnnPref(chanPoint *wire.OutPoint,
+	pref bool) error {
+	return f.cfg.Wallet.Cfg.Database.Update(func(tx *bolt.Tx) error {
+
+		bucket, err := tx.CreateBucketIfNotExists(openChanAnnPrefBucket)
+		if err != nil {
+			return err
+		}
+
+		var outpointBytes bytes.Buffer
+		if err = writeOutpoint(&outpointBytes, chanPoint); err != nil {
+			return err
+		}
+
+		scratch := make([]byte, 0)
+		var b byte
+		if pref {
+			b = 1
+		} else {
+			b = 0
+		}
+		scratch = append(scratch[:], b)
+
+		return bucket.Put(outpointBytes.Bytes(), scratch[:])
+	})
+}
+
+// getOpenChanAnnPref retrives an open channel's announcement preference.
+func (f *fundingManager) getOpenChanAnnPref(chanPoint *wire.OutPoint) (bool, error) {
+
+	var pref bool
+	err := f.cfg.Wallet.Cfg.Database.View(func(tx *bolt.Tx) error {
+
+		bucket := tx.Bucket(openChanAnnPrefBucket)
+		if bucket == nil {
+			return fmt.Errorf("Channel announcement preference " +
+				"not found")
+		}
+
+		var outpointBytes bytes.Buffer
+		if err := writeOutpoint(&outpointBytes, chanPoint); err != nil {
+			return err
+		}
+
+		value := bucket.Get(outpointBytes.Bytes())
+		if value == nil {
+			return fmt.Errorf("Channel announcement preference " +
+				"not found")
+		}
+
+		if value[0] == 1 {
```

Comment:
[binary.Read](https://golang.org/pkg/encoding/binary/#Read) can be used here as it now supports reading boolean arguments values.